adoc/Private-Registry-Airgap.adoc

```diff
@@ -0,0 +1,247 @@
+== Deployment In Air Gap Environment 
+
+=== Introduction
+
+K3s is a lightweight Kubernetes distribution from Rancher that can be
+deployed on a single node. 
+
+=== Use Cases
+
+Testing:: The customer wants to test SUSE Private Registry, but they do not
+have a Kubernetes cluster prepared.
+Single-node setup:: A production setup for customers that want to use SUSE
+Private Registry, but do not want to deploy CaaSP, Racher's RKE or any other
+Kubernetes cluster.
+Air-gapped environments:: K3s can be deployed without access to the Internet.
+With the SUSE Private Registry running on such a cluster, it could for example
+serve as an image registry for other Kubernetes clusters in the customer's
+network.
+
+=== Requirements
+
+You should have Helm installed on the machine where you want to install k3s.
+
+Make sure that you have enough disk space. Rancher does not mention any specific
+requirements for disk size, so prepare at least a few gigabytes. It is possible
+to put the `/var/lib/rancher` directory on any partition or disk where you know
+that you have enough space. In case space starts to run out, you might encounter
+https://github.com/rancher/k3s/issues/1552[k3s issue #1552].
+
+=== Deployment
+
+. Deploy k3s into an air-gapped environment by following
+https://rancher.com/docs/k3s/latest/en/installation/airgap/#manually-deploy-images-method[the upstream guide].
+
+.. You will need the k3s binary from: https://github.com/rancher/k3s/releases
++
+There is a list of images required for basic k3s functionality and for some
+optional features (such as the Traefik ingress controller) here:
+https://github.com/rancher/k3s/releases/download/v1.18.10%2Bk3s1/k3s-airgap-images-amd64.tar[k3s-airgap-images-amd64.tar]
++
+Download both these files and transfer them to your air-gapped machine.
+
+.. On the machine where k3s should run, prepare the images and install
+the binary:
++
+[source,bash]
+----
+sudo mkdir -p /var/lib/rancher/k3s/agent/images/
+sudo cp k3s-airgap-images-amd64.tar /var/lib/rancher/k3s/agent/images/
+sudo cp k3s /usr/local/bin/k3s
+sudo chown +x /usr/local/bin/k3s
+----
+
+.. Install k3s. Do not start the service yet.
++
+[source,bash]
+----
+INSTALL_K3S_SKIP_ENABLE=true INSTALL_K3S_SKIP_DOWNLOAD=true ./install.sh
+----
+
+.. Fetch the images required for the SUSE Private Registry and install them
```

Comment:
As you can see, these images are downloaded from the internet. So it might be better to move the section where the images are downloaded more up, before we move to the airgapped machine...

Reply:
1. Basically so that the parts where one is fetching stuff from internet goes first and then goes the other part when working in airgap environment. In the original text this is kind of mixed (first k3s stuff download + install, then harbor images, download + copy).

2. Yes, that's a good idea, point to the latest release, or just release that user wants to use.

Reply:
I have added a mention of checking, and I've updated the URL to point to `v1.18.13+k3s1`. Since I started this, it looks like v1.19 has been released too, but I do not know if k3s uses a minor_ver=even → stable, minor_ver=odd → unstable system or anything like that.

I have not otherwise disentangled the installation steps, as I have not installed this myself and do not have a suitable machine. 

My understanding of air-gapped deployment means that the host machine has no internet connection, so I do not know how the step that does `docker pull` in a `do` loop will work. Perhaps I misunderstand what air-gapped means in this context. 

@jsuchome, if you want the steps disentangled, could you please sent a PR or at least give me new numbered steps? I know nothing about k3s, SPRpbH or anything to do with it. I had never heard of them before I was asked to migrate your Howto from Confluence.

Reply:
_My understanding of air-gapped deployment means that the host machine has no internet connection, so I do not know how the step that does docker pull in a do loop will work. Perhaps I misunderstand what air-gapped means in this context._

That was exactly my point - and now you can see that the original document is poorly written when it was not clear from it. So, the download parts need to be done on a machine with internet access, that's why I'm suggesting to move `docker pull` commands up to the top where you are also downloading `k3s` binary and installation script. 

We're saying _Transfer both files to your air-gapped machine._ about k3s above, so we should move the `docker pull` commands before this, and then explain something like _transfer k3s files and harbor docker images to your air-gapped machine_

Reply:
About the version: I would not care that much, linking specific version from this document - after all, the link becomes obsolete when they release a new one. Just say to fetch the latest stable version from the list of releases...

Reply:
OK, done. I have said use the latest stable version, linked to the releases page https://github.com/k3s-io/k3s/releases and the latest-release page https://github.com/k3s-io/k3s/releases/latest and removed the reference to any specific version. 

I have removed the "for example" mention when downloading the install script, and specified the filename for the `k3s` binary.